pkg/operators/oci-handler/extrainfo.go

```diff
@@ -44,6 +44,22 @@ func addExtraInfo(gadgetCtx operators.GadgetContext, metadata []byte, manifest *
 
 	created := manifest.Annotations[ocispec.AnnotationCreated]
 
+	datasources := []*api.DataSource{}
+	for _, ds := range gadgetCtx.GetDataSources() {
+		datasource := &api.DataSource{
+			Name: ds.Name(),
+		}
+		datasource.Fields = make([]*api.Field, len(ds.Fields()))
+		for i, field := range ds.Fields() {
+			datasource.Fields[i] = &api.Field{
+				Name:        field.Name,
+				Annotations: map[string]string{"description": field.Annotations["description"]},
+			}
+		}
+		datasources = append(datasources, datasource)
+	}
+	datasourcesJson, _ := json.Marshal(datasources)
```

Comment:
You can marshal `GetDataSources` directly. In this way, the missing fields would be automatically added.
```suggestion
	datasourcesJson, _ := json.Marshal(gadgetCtx.GetDataSources())
```